lib/seed/runner.rb

```diff
@@ -0,0 +1,144 @@
+      hsh = sum_facility_totals
+      total_counts.merge!(hsh)
+
+      msg = "⭐️  Seed complete! Elasped time #{distance_of_time_in_words(start_time, Time.current, include_seconds: true)} ⭐️"
+      puts msg
+      logger.info msg: msg, counts: counts
+      [counts, total_counts]
+    end
+
+    def seed_patients(progress)
+      parallel_options = {
+        finish: lambda do |item, i, result|
+          slug, facility_size = item[1], item[2]
+          progress.log("Finished facility: [#{slug}, #{facility_size}] counts: #{result}")
+          progress.increment
+        end
+      }
+      parallel_options[:in_processes] = 0 if Rails.env.test?
+
+      facility_info = Facility.pluck(:id, :slug, :facility_size)
+      Parallel.map(facility_info, parallel_options) do |(facility_id, slug, facility_size)|
+        benchmark("Seeding records for facility #{slug}") do
+          result = {facility: slug}
+          facility = Facility.find(facility_id)
+          user = facility.users.find_by!(role: config.seed_generated_active_user_role)
+          # Set a "birth date" for the Facility that patient records will be based from
+          facility_birth_date = Faker::Time.between(from: 3.years.ago, to: 1.day.ago)
+          benchmark("[#{slug} Seeding patients for a #{facility_size} facility") do
+            patients = patients_to_create(facility_size).times.map { |num|
+              build_patient(user, oldest_registration: facility_birth_date)
+            }
+            patient_result = Patient.import(patients, recursive: true)
+            result[:patient] = patient_result.ids.size
+          end
+          patient_info = facility.assigned_patients.pluck(:id, :recorded_at)
+          bp_result = BloodPressureSeeder.call(config: config, facility: facility, user: user)
+          result.merge! bp_result
+          appt_result = create_appts(patient_info, user)
+          result[:appointment] = appt_result.ids.size
+          result
+        end
+      end
+    end
+
+    def self.random_gender
+      return Patient::GENDERS.sample if Patient::GENDERS.size == 2
+      num = rand(100)
+      if num <= 1
+        :transgender
+      elsif num > 1 && num < 50
+        :male
+      else
+        :female
+      end
+    end
+
+    def patients_to_create(facility_size)
+      max = config.max_patients_to_create.fetch(facility_size.to_sym)
+      config.rand_or_max((0..max), scale: true).to_i
+    end
+
+    def sum_facility_totals
+      counts.each_with_object(Hash.new(0)) { |(_slug, counts), hsh| counts.each { |type, count| hsh[type] += count } }
+    end
+
+    def build_patient(user, oldest_registration:)
+      recorded_at = Faker::Time.between(from: oldest_registration, to: 1.day.ago)
+      FactoryBot.build(:patient,
+        address: nil,
+        recorded_at: recorded_at,
+        registration_user: user,
+        registration_facility: user.facility)
+    end
+
+    def create_appts(patient_info, user)
```

Comment:
Can we tie appointments to the patient's BPs/encounters rather than generate them separately? We can keep the `some patients dont get appointments` behaviour

That way we'd end up with a more organic pattern of visits -> follow up appointments. Scheduled date should tie into a previous encounter's creation date so we have some patients going overdue, LTFU etc when we seed this data.